File: oops/Object-Oriented-Design-main/pattern technique/News_Feed_System/News_Feed_System/src/com/company/Model/Post.java
package News_Feed_System.src.com.company.Model;

import java.util.List;

public class Post {
    
    private int id;
    private int userId;
    private int upVote;
    private int downVote;
    private String timeStamp;
    private String content;
    private List<Comment> commentsList;
    private String author;
    
    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public int getUserId() {
        return userId;
    }
    public void setUserId(int userId) {
        this.userId = userId;
    }
    public String getTimeStamp() {
        return timeStamp;
    }
    public void setTimeStamp(String timeStamp) {
        this.timeStamp = timeStamp;
    }
    public String getContent() {
        return content;
    }
    public void setContent(String content) {
        this.content = content;
    }
    public List<Comment> getCommentsList() {
        return commentsList;
    }
    public void setCommentsList(List<Comment> commentsList) {
        this.commentsList = commentsList;
    }
    public int getUpVote() {
        return this.upVote;
    }
    public void setUpVote() {
        this.upVote++;
    }
    public int getDownVote() {
        return this.downVote;
    }
    public void setDownVote() {
        this.downVote++;
    }
    public String getAuthor() {
        return author;
    }
    public void setAuthor(String author) {
        this.author = author;
    }
    
    
}

File: oops/Object-Oriented-Design-main/pattern technique/Harish_BonusCode/models/post.py
from datetime import datetime

class UserPost:

    def __init__(self, id, userId, data) -> None:
        self.__id = id
        self.__data = data
        self.__userId = userId
        self.__upVote = 0
        self.__downVote = 0
        self.datetime = str(datetime.now())

    def getId(self):
        return self.__id

    def getData(self):
        return self.__data

    def getUpvote(self):
        return self.__upVote

    def getDownVote(self):
        return self.__downVote

    def getUserId(self):
        return self.__userId

    def getDate(self):
        return self.datetime

    def updateUpVote(self):
        self.__upVote+=1
    
    def updateDownVote(self):
        self.__downVote+=1

